pkg/controller/plan/adapter/converter.go

```diff
@@ -99,42 +95,62 @@ func (c *Converter) ConvertPVCs(pvcs []*v1.PersistentVolumeClaim, srcFormat srcF
 	return false, nil
 }
 
-func (c *Converter) ensureJob(pvc *v1.PersistentVolumeClaim, srcFormat, dstFormat string) (*batchv1.Job, error) {
-	jobName := getJobName(pvc, "convert")
-	job := &batchv1.Job{}
-	err := c.Destination.Client.Get(context.Background(), client.ObjectKey{Name: jobName, Namespace: pvc.Namespace}, job)
+func (c *Converter) deleteScratchDV(scratchDV *cdi.DataVolume) {
+	err := c.Destination.Client.Delete(context.Background(), scratchDV, &client.DeleteOptions{})
+	if err != nil {
+		c.Log.Error(err, "Failed to delete scratch DV", "DV", scratchDV.Name)
+	}
+}
+
+func (c *Converter) ensureJob(pvc *v1.PersistentVolumeClaim, dv *cdi.DataVolume, srcFormat, dstFormat string) (*batchv1.Job, error) {
+	// Find existing job by label
+	jobList := &batchv1.JobList{}
+	label := client.MatchingLabels{planbase.AnnConversionSourcePVC: pvc.Name}
+	err := c.Destination.Client.List(context.Background(), jobList, client.InNamespace(pvc.Namespace), label)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(jobList.Items) == 1 {
+		c.Log.Info("Found convert job", "job", jobList.Items[0].Name)
+		return &jobList.Items[0], nil
+	} else if len(jobList.Items) > 1 {
+		return nil, liberr.New("multiple convert jobs found for pvc", "pvc", pvc.Name)
+	}
+
+	// Job doesn't exist, create it
+	job := createConvertJob(pvc, dv, srcFormat, dstFormat, c.Labels)
+	c.Log.Info("Creating convert job", "pvc", pvc.Name, "srcFormat", srcFormat, "dstFormat", dstFormat)
+	err = c.Destination.Client.Create(context.Background(), job, &client.CreateOptions{})
 	if err != nil {
-		if k8serr.IsNotFound(err) {
-			c.Log.Info("Converting PVC format", "pvc", pvc.Name, "srcFormat", srcFormat, "dstFormat", dstFormat)
-			job := createConvertJob(pvc, srcFormat, dstFormat, c.Labels)
-			err = c.Destination.Client.Create(context.Background(), job, &client.CreateOptions{})
-			if err != nil {
-				return nil, err
-			}
-		}
 		return nil, err
 	}
 
-	return job, err
+	return job, nil
 }
 
-func createConvertJob(pvc *v1.PersistentVolumeClaim, srcFormat, dstFormat string, labels map[string]string) *batchv1.Job {
+func createConvertJob(pvc *v1.PersistentVolumeClaim, dv *cdi.DataVolume, srcFormat, dstFormat string, labels map[string]string) *batchv1.Job {
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	labels[planbase.AnnConversionSourcePVC] = pvc.Name
 	return &batchv1.Job{
 		ObjectMeta: meta.ObjectMeta{
-			Name:      fmt.Sprintf("convert-%s", pvc.Name),
-			Namespace: pvc.Namespace,
-			Labels:    labels,
+			GenerateName: fmt.Sprintf("convert-%s-", pvc.Name),
+			Namespace:    pvc.Namespace,
+			Labels:       labels,
 		},
 		Spec: batchv1.JobSpec{
 			BackoffLimit: ptr.To(int32(3)),
 			Completions:  ptr.To(int32(1)),
 			Template: v1.PodTemplateSpec{
 				Spec: v1.PodSpec{
 					SecurityContext: &v1.PodSecurityContext{
+						RunAsNonRoot: ptr.To(true),
```

Comment:
how about adding a comment about the 'default' namespace, something like "specifying RunAsNonRoot=true without RunAsUser would fail the creation of the pod when migrating to the namespace 'default' in OpenShift but even if the pod starts, its execution would fail in this case"

Reply:
cool thanks, filed https://issues.redhat.com/browse/MTV-964 to remember to document it as a known issue